src/__mocks__/graceful-fs.js

```diff
@@ -91,28 +91,62 @@ fs.stat.mockImpl(function(filepath, callback) {
 
   if (node && typeof node === 'object') {
     callback(null, {
-      isDirectory: function() {
-        return true;
-      },
-      isSymbolicLink: function() {
-        return false;
-      },
-      mtime: mtime,
+      isDirectory: () => true,
+      isSymbolicLink: () => false,
+      mtime,
     });
   } else {
     callback(null, {
-      isDirectory: function() {
-        return false;
-      },
-      isSymbolicLink: function() {
-        return false;
-      },
-      mtime: mtime,
+      isDirectory: () => false,
+      isSymbolicLink: () => false,
+      mtime,
     });
   }
 });
 
-const noop = () => {};
+fs.statSync.mockImpl((filepath) => {
+  const node = getToNode(filepath);
+
+  if (node.SYMLINK) {
+    return fs.statSync(node.SYMLINK);
```

Comment:
I think this solution is fine, I just meant to comment on your use of recursion :)